plugins/modules/ntnx_volume_groups.py

```diff
@@ -0,0 +1,413 @@
+            volume_group_uuid, endpoint="/iscsi-client-attachments"
+        )
+        result["response"]["clients"] = clients.get("data")
+
+
+# def update_volume_group(module, result):
+#     volume_group = VolumeGroup(module)
+#     volume_group_uuid = module.params.get("volume_group_uuid")
+#     if not volume_group_uuid:
+#         result["error"] = "Missing parameter volume_group_uuid in playbook"
+#         module.fail_json(msg="Failed updating volume_group", **result)
+#     result["volume_group_uuid"] = volume_group_uuid
+#
+#     # read the current state of volume_group
+#     resp = volume_group.read(volume_group_uuid)
+#     resp = resp.get("volume_group")
+#
+#     # new spec for updating volume_group
+#     update_spec, error = volume_group.get_spec(resp)
+#     if error:
+#         result["error"] = error
+#         module.fail_json(msg="Failed generating volume_group update spec", **result)
+#
+#     # check for idempotency
+#     if resp == update_spec:
+#         result["skipped"] = True
+#         module.exit_json(
+#             msg="Nothing to change. Refer docs to check for fields which can be updated"
+#         )
+#
+#     if module.check_mode:
+#         result["response"] = update_spec
+#         return
+#
+#     # update volume_group
+#     volume_group.update(update_spec, uuid=volume_group_uuid, no_response=True)
+#
+#     resp = volume_group.read(volume_group_uuid)
+#
+#     result["changed"] = True
+#     result["response"] = resp
+
+
+def delete_volume_group(module, result):
+    def detach_iscsi_clients():
+        clients_resp = volume_group.read(
+            volume_group_uuid, endpoint="/iscsi-client-attachments"
+        )
+        detached_clients = []
+        for client in clients_resp.get("data", []):
+
+            detach_resp = volume_group.detach_iscsi_client(volume_group_uuid, client)
+
+            task_uuid = detach_resp["data"]["extId"][-36:]
+            wait_for_task_completion(module, {"task_uuid": task_uuid})
+            detached_clients.append(client["extId"])
+
+        result["detached_clients"] = detached_clients
+
+    def detach_vms():
+        vms_resp = volume_group.read(volume_group_uuid, endpoint="/vm-attachments")
+        detached_vms = []
+        for vm in vms_resp.get("data", []):
+
+            detach_resp = volume_group.detach_vm(volume_group_uuid, vm)
+
+            task_uuid = detach_resp["task_uuid"]
+            wait_for_task_completion(module, {"task_uuid": task_uuid})
+            detached_vms.append(vm["extId"])
+
+        result["detached_vms"] = detached_vms
+
+    volume_group_uuid = module.params["volume_group_uuid"]
+    if not volume_group_uuid:
+        result["error"] = "Missing parameter volume_group_uuid in playbook"
+        module.fail_json(msg="Failed deleting volume_groups", **result)
+
+    volume_group = VolumeGroup(module)
+
+    detach_iscsi_clients()
+    detach_vms()
+
+    resp = volume_group.delete(volume_group_uuid)
+    resp.pop("metadata")
```

Comment:
why removing metadata ?

Reply:
cause there is no clear information for user